crates/environ/src/builtin.rs

```diff
@@ -45,6 +45,18 @@ macro_rules! foreach_builtin_function {
             externref_global_get(vmctx, i32) -> (reference);
             /// Returns an index for Wasm's `global.get` instruction for `externref`s.
             externref_global_set(vmctx, i32, reference) -> ();
+            /// Returns an index for wasm's `memory.atomic.notify` for locally defined memories.
+            memory_atomic_notify(vmctx, i32, i32, i32) -> (i32);
+            /// Returns an index for wasm's `memory.atomic.notify` for imported memories.
+            imported_memory_atomic_notify(vmctx, i32, i32, i32) -> (i32);
```

Comment:
Due to the relative cost of these functions, could the distinction between local/imported memories be removed to halve the number of intrinsics added?

Reply:
I agree, though I adapted a pattern used by `imported_memory_fill`/ `imemory_fill` or `imported_memory32_grow`/`memory32_grow`. Are there plans to do the same with above?

Reply:
Eventually I'd like to get to those and do the same yeah, we did it awhile ago for a few reference types intrinsics but haven't gotten around to fixing things generally.

Reply:
Opened https://github.com/bytecodealliance/wasmtime/issues/2561 to address it in the future.